native_client/ctcdecode/ctc_beam_search_decoder.cpp

```diff
@@ -57,6 +57,12 @@ DecoderState::init(const Alphabet& alphabet,
   return 0;
 }
 
+void
+DecoderState::withLogits(bool val)
+{
+  keep_logits_ = val;
+}
+
```

Comment:
We should set this in `::init`, we don't want to have to reason about already initialized `DecoderState`s which then enable saving of logits.

Reply:
That's fine, better even I just didn't want to add another parameter without checking first. I'll change that now.

Reply:
Ah, now I remember why there was a problem here...

The decoder is initialised in

```c
STT_CreateStream(ModelState* aCtx,
                StreamingState** retval)
```

I didn't want to add another parameter to an API function, or proliferate new API calls. 

A couple of options are:
1. Add a `STT_CreateStreamWithLogits()` API call.
2. Change the `STT_CreateStream()` to include a `bool keep_logits` parameter.
3. Add `keep_logits_` variable to `ModelState` (we had this before, but I include it for completeness)

Any options I've missed?

Reply:
I'd go with:

> 1. Add a STT_CreateStreamWithLogits() API call.

But not an API call, just a private abstraction. See for example `CreateStreamAndFeedAudioContent`.